libdnf/rpm/nevra.cpp

```diff
@@ -202,8 +202,8 @@ std::vector<Nevra> Nevra::parse(const std::string & nevra_str, const std::vector
                 ++arch_delim_for_nev;
 
                 // test arch for absence of '-'
-                if (evr_delim != nullptr) {
-                    const char * evr_delim_for_na = release_delim == nullptr ? evr_delim : release_delim;
+                if (before_last_delim != nullptr) {
+                    const char * evr_delim_for_na = last_delim == nullptr ? before_last_delim : last_delim;
```

Comment:
Strange. The bug must have been here in the previous version.
If `before_last_delim != nullptr` then `last_delim != nullptr`. The condition does not make sense.
And eg `name.ar-ch` is not captured by the test.

I suggest adding more unit tests to NA: `name.ar-ch` and `name.-arch` and `name.`

And here is an example parsing of NA with fixed `-` detection in the `arch` and optimization.
```
            case Form::NA: {
                if (arch_delim == nullptr) {
                    continue;
                }
                // test: NA cannot contain ':'
                if (epoch_delim != nullptr) {
                    continue;
                }

                // test name presence
                if (arch_delim == nevra_pattern) {
                    continue;
                }

                // test arch for absence of '-'
                if (last_delim != nullptr && last_delim > arch_delim) {
                    continue;
                }

                // test arch presence
                if (end - arch_delim == 1) {
                    continue;
                }

                nevra.name.assign(nevra_pattern, arch_delim);
                nevra.arch.assign(arch_delim + 1);

                result.push_back(std::move(nevra));
            } break;
```